tests/cephfs/cephfs_bugs/evict_all_clients_add_to_blocklist.py

```diff
@@ -0,0 +1,147 @@
+import random
+import string
+import traceback
+
+from tests.cephfs.cephfs_utilsV1 import FsUtils
+from utility.log import Log
+
+log = Log(__name__)
+
+"""
+ceph --admin-daemon $socketfile client evict [-h|--help] evicts ALL clients.
+It is observed that adding "--help|-h" with the "client evict" command
+terminates all active client connections and adds to the blocklist.
+ceph tell DAEMON_NAME client evict id=ID_NUMBER
+Steps to Reproduce:
+1. Connect clients to cephfs server
+2. # ceph --admin-daemon $socketfile client evict [-h|--help] evicts ALL clients
+3. Observe all active sessions using cephfs are lost and
+4. Observe all active clients have been added to blocklist
+5. Remove the clients from blocklist using "ceph osd blocklist rm IP_ADDRESS"
+6. Verify the clients are removed from blocklist
+"""
+
+
+def run(ceph_cluster, **kw):
+    try:
+        tc = "58619"
+        log.info(f"Running CephFS tests for ceph tracker - {tc}")
+        # Initialize the utility class for CephFS
+        fs_util = FsUtils(ceph_cluster)
+        # Get the client nodes
+        clients = ceph_cluster.get_ceph_objects("client")
+        config = kw.get("config")
+        # Authenticate the clients
+        fs_util.auth_list(clients)
+        build = config.get("build", config.get("rhbuild"))
+        # Prepare the clients
+        fs_util.prepare_clients(clients, build)
+        client1 = clients[0]
+        fs_details = fs_util.get_fs_info(client1)
+        if not fs_details:
+            fs_util.create_fs(client1, "cephfs")
+        # Check if jq package is installed, install if not
+        jq_check = client1.exec_command(
+            sudo=True, cmd="rpm -qa | grep jq", check_ec=False
+        )
+        if "jq" not in jq_check:
+            client1.exec_command(sudo=True, cmd="yum install -y jq")
+        # Generate random string for directory names
+        rand = "".join(
+            random.choice(string.ascii_lowercase + string.digits) for _ in range(5)
+        )
+        # Define mount directories
+        fuse_mounting_dir_1 = f"/mnt/cephfs_fuse_{rand}"
+        fuse_mounting_dir_2 = f"/mnt/cephfs_fuse_{rand}_2"
+        fuse_mounting_dir_3 = f"/mnt/cephfs_fuse_{rand}_3"
+        kernel_mounting_dir_1 = f"/mnt/cephfs_kernel_{rand}"
+        kernel_mounting_dir_2 = f"/mnt/cephfs_kernel_{rand}_2"
+        kernel_mounting_dir_3 = f"/mnt/cephfs_kernel_{rand}_3"
+        # Mount CephFS using ceph-fuse and kernel
+        fs_util.fuse_mount([client1], fuse_mounting_dir_1)
+        fs_util.fuse_mount([client1], fuse_mounting_dir_2)
+        fs_util.fuse_mount([client1], fuse_mounting_dir_3)
+        mon_node_ips = fs_util.get_mon_node_ips()
+        fs_util.kernel_mount([client1], kernel_mounting_dir_1, ",".join(mon_node_ips))
+        fs_util.kernel_mount([client1], kernel_mounting_dir_2, ",".join(mon_node_ips))
+        fs_util.kernel_mount([client1], kernel_mounting_dir_3, ",".join(mon_node_ips))
+        # get client ID for rank 0 MDS
```

Comment:
It would be good to add some IO for it to be in-progress while clients are evicted.

Reply:
IOs added